tests/trainer/test_checkpoint.py

```diff
@@ -507,290 +186,333 @@ def test_checkpoint_with_object_store_logger(
-    first_trainer_final_checkpoint_filepath = os.path.join(rank_to_checkpoint_a_folder[0], final_checkpoint)
-
-    # Move the resume and final file to the rank 0 folder
-    try:
-        rank_checkpoint_filepath = os.path.join(checkpoint_a_folder, resume_file.format(rank=dist.get_global_rank()))
-        shutil.copy2(rank_checkpoint_filepath,
-                     checkpoint_to_resume_filepath.format(rank=dist.get_global_rank()),
-                     follow_symlinks=True)
-    except (shutil.SameFileError, FileNotFoundError):
-        pass
-
-    try:
-        rank_checkpoint_filepath = os.path.join(checkpoint_a_folder,
-                                                final_checkpoint.format(rank=dist.get_global_rank()))
-        shutil.copy2(rank_checkpoint_filepath,
-                     first_trainer_final_checkpoint_filepath.format(rank=dist.get_global_rank()),
-                     follow_symlinks=True)
-    except (shutil.SameFileError, FileNotFoundError):
-        pass
-
-    checkpoint_b_folder = os.path.join(rank_to_checkpoint_a_folder[0], 'second')
-
-    second_trainer_hparams.save_folder = checkpoint_b_folder
-    second_trainer_hparams.load_path = checkpoint_to_resume_filepath
-    second_trainer_hparams.load_weights_only = False
-    second_trainer_hparams.load_strict_model_weights = False
-
-    _test_checkpoint_trainer(second_trainer_hparams)
-    dist.barrier()  # Ensure all ranks wrote the checkpoint file
-    second_trainer_final_checkpoint_filepath = os.path.join(checkpoint_b_folder, final_checkpoint)
-
-    assert_checkpoints_equivalent(
-        checkpoint_file_a=first_trainer_final_checkpoint_filepath,
-        checkpoint_file_b=second_trainer_final_checkpoint_filepath,
+        # TODO (mihir): should be checking the entire state
+        # not just the model?
+        self._assert_weights_equivalent(
+            trainer_1.state.model,
+            trainer_2.state.model,
+        )
+
+        assert trainer_1.state.run_name == trainer_2.state.run_name
+
+    def test_different_run_names(self):
+
+        trainer_1 = self.get_trainer(
+            save_folder='first/',
+            seed=12345,
+        )
+        trainer_1.fit()
+        trainer_1.close()
+
+        trainer_2 = self.get_trainer(
+            load_path=os.path.join('first', 'ep2.pt'),
+            seed=67890,
+        )
+
+        assert trainer_1.state.run_name != trainer_2.state.run_name
+
+    @device('cpu', 'gpu')
+    @pytest.mark.parametrize('save_overwrite', [True, False])
+    def test_save_overwrite(self, device, save_overwrite):
+
+        trainer_1 = self.get_trainer(
+            save_folder='first',
+            device=device,
+        )
+        trainer_1.fit()
+        trainer_1.close()
+
+        if save_overwrite:
+            ctx = contextlib.nullcontext
+        else:
+            ctx = functools.partial(pytest.raises, FileExistsError)
+
+        with ctx():  # expect FileExistsError if save_overwrite=False
```

Comment:
(minor) nit: would prefer to one-line or define `ctx = None` at top. Surprised lint isn't complaining about this.

Reply:
Since it's an `if .., else`, the `ctx = None` would always be overridden, so it should be ok?

Reply:
My personal preference is to always put a `ctx=None` even if its always overridden to make scope clear. Even though python doesn't scope indents, most people tend to think in those scope levels, so its nice to be explicit about it. With that said, this is a minor nit you are free to ignore without further discussion :)

Reply:
OK understood, added the `ctx = None`.